src/app/panels/image-panel/image-panel.component.ts

```diff
@@ -12,24 +11,74 @@ import { EVTModelService } from '../../services/evt-model.service';
 export class ImagePanelComponent {
   @Input() viewerData: ViewerDataType;
 
-  currentMsDescId$ = new BehaviorSubject(undefined);
-  currentMsDesc$ = combineLatest([this.evtModelService.msDesc$, this.currentMsDescId$]).pipe(
-    filter(([msDesc, currentId]) => !!msDesc && !!currentId),
-    map(([msDesc, currentId]) => msDesc.find(m => m.id === currentId)),
-  );
-
-  msDescOpen = false;
+  get imageIndex() { return this._imageIndex; }
 
   constructor(
-    private evtModelService: EVTModelService,
+    public evtModelService: EVTModelService,
+    public http: HttpClient,
+    private cdref: ChangeDetectorRef,
   ) {
   }
+  public msDesc$ = this.evtModelService.msDesc$;
+  public showSecondaryContent = false;
+  public selectedPage;
+  public msDescID = '';
+  public value: number;
+  public images = [];
+  public currentImg: string;
+  // tslint:disable-next-line: variable-name
+  private _imageIndex: number;
+  public manifest;
+  public imagePath;
+  public filename;
+
+  toggleImg(event){
+    this._imageIndex = this.images.indexOf(event);
+  }
+
+  eventPageImg(pageImg: number){
+    if (this.viewerData?.type === 'manifest'){
+      this.currentImg = this.images[pageImg - 1].label;
+    }
+    if (this.viewerData?.type === 'default' || this.viewerData?.type === 'graphics'){
+      this.currentImg = this.images[pageImg - 1].url.split('/').pop().split('.')[0];
+    }
+    this.cdref.detectChanges();
+  }
 
-  setMsDescOpen(isOpen: boolean) {
-    this.msDescOpen = isOpen;
+  // tslint:disable-next-line: use-lifecycle-interface
+  ngOnChanges() {
+    if (this.viewerData?.type === 'manifest'){
+      this.manifest = this.viewerData.value?.manifestURL;
+      // tslint:disable-next-line: no-any
+      this.http.get(`${this.manifest}`).subscribe((item: any) => {
+        item.sequences.forEach((value) => {
+          this.images.push(...value.canvases);
+        });
+        this.currentImg = this.images[0].label;
+      });
+    }
+    if (this.viewerData?.type === 'default' || this.viewerData?.type === 'graphics'){
+      this.images = this.viewerData?.value?.xmlImages;
+      this.imagePath =  this.viewerData?.value?.xmlImages[0]?.url;
+      this.filename = this.imagePath.split('/').pop().split('.')[0];
+      this.currentImg = this.filename;
+    }
+    else {
+      this.currentImg = 'Error loading';
+    }
   }
```

Comment:
All the logic here depends on `viewerData`. You can try to change the `@Input` prop into a setter/getter and move this logic inside it.
```
private _vd: ViewerDataType;
@Input() set viewerData(vd: ViewerDataType) {
   // Your logic goes here
}
get viewerData() {
   return this._vd;
}
```
NB: The getter is only needed if you access `viewerData` elsewhere, otherwise you can omit it.